src/main/java/com/wanderersoftherift/wotr/datagen/ModRiftObjectiveLootTableProvider.java

```diff
@@ -0,0 +1,77 @@
+package com.wanderersoftherift.wotr.datagen;
+
+import com.wanderersoftherift.wotr.WanderersOfTheRift;
+import com.wanderersoftherift.wotr.init.ModItems;
+import net.minecraft.core.HolderLookup;
+import net.minecraft.core.registries.Registries;
+import net.minecraft.data.loot.LootTableSubProvider;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.level.storage.loot.LootPool;
+import net.minecraft.world.level.storage.loot.LootTable;
+import net.minecraft.world.level.storage.loot.entries.LootItem;
+import net.minecraft.world.level.storage.loot.functions.SetItemCountFunction;
+import net.minecraft.world.level.storage.loot.functions.SetNameFunction;
+import net.minecraft.world.level.storage.loot.providers.number.BinomialDistributionGenerator;
+import net.minecraft.world.level.storage.loot.providers.number.ConstantValue;
+import net.minecraft.world.level.storage.loot.providers.number.UniformGenerator;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.function.BiConsumer;
+
+public record ModRiftObjectiveLootTableProvider(HolderLookup.Provider registries) implements LootTableSubProvider {
+
+    public void generate(BiConsumer<ResourceKey<LootTable>, LootTable.Builder> consumer) {
+        consumer.accept(getResourceKey("rift_objective/success"), LootTable.lootTable()
```

Comment:
We should also add the runegem geodes to this. Just because we can. That loottable exists elsewhere though.

Might have an issue with checking the rift tier.

Reply:
The player will be in the overworld by this point, right. But the rift tier is available, and could be added as a loot context param. And then the rift tier check can try to grab the from the loot context first and level second?